### ISSUE TITLE
GRBL 0.9j proeblem, not activate buttons

### ISSUE THREAD

**Issue by luisluna2** [cncjs/cncjs#384]
CNCjs 1.9.16-pre-896

Buttons of jogginf and spindle Not working when connect

![image](https://user-images.githubusercontent.com/38824145/45718067-77be9280-bb61-11e8-9b85-163f6b47d28b.png)

**Comment by luisluna2** [comment]
> Your Grbl controller does not respond correctly. Try disabling "Enable hardware flow control" before connecting, or clicking the Reset button and check the message in the console widget.

Hello friend, I did what he told me, but it does not work.

**Comment by cheton** [comment]
@luisluna2 You can connect to Grbl via Arduino serial monitor. If everything is ok, a welcome message will pop up `Grbl 0.9j ['$' for help]`, and you can see status output by sending the "?" command.

![image](https://user-images.githubusercontent.com/447801/45990337-5b1ae100-c0b2-11e8-9ab2-a69ba86cd13e.png)

If you cannot make it work via Arduino IDE, try upgrading your Grbl firmware from 0.9j to 1.1f.
https://github.com/gnea/grbl/releases
https://github.com/gnea/grbl/wiki/Compiling-Grbl

**Comment by luisluna2** [comment]
@cheton Thank you Friend, I will check this tomorrow and I will reply.

**Comment by luisluna2** [comment]
hello check this:

![image](https://user-images.githubusercontent.com/38824145/46044426-53723100-c0e0-11e8-8a56-c40979d2bb50.png)


and same:

![image](https://user-images.githubusercontent.com/38824145/46044463-6c7ae200-c0e0-11e8-91aa-9e35c113a7ad.png)

**Comment by MitchBradley** [comment]
Okay, it looks like you have limit switches enabled in your GRBL configuration.  There are several possibilities:

1. You actually have limit switches and they are active, i.e. the machine is already at its limit position
2. You have limit switches and they are misconfigured - the polarity is wrong so they are saying the machine is at the limit when it is not
3. You do not have limit switches but your GRBL configuration is set up as if you did have them - so GRBL is trying to sense switches that do not exist and it thinks they are active

You can turn off GRBL limits by typing the following command in the serial monitor window:

`$21=0`

If you do have limit switches but their polarity is wrong, you can change the polarity with

`$5=1`

or 

`$5=0`

$5=0 is for the default behavior where pin low means limit triggered.  $5=1 is for the opposite behavior - pin high is limit triggered.

**Comment by luisluna2** [comment]
@MitchBradley I try the 3 options and the same, my GRBL is 0.9j maybe with new 1.1 can work, I'm checking now...

**Comment by MitchBradley** [comment]
Do you have soft limits configured?

**Comment by luisluna2** [comment]
@MitchBradley  yes friend, but only I have a arduino for check the program, not connect in my CNC, but I check with 1.1f and all is okay!, I will change of GRBL version.
